packages/components/src/tooltip/index.native.js

```diff
@@ -54,11 +55,12 @@ const useKeyboardVisibility = () => {
 			ios: 'keyboardWillHide',
 		} );
 		const hideListener = Keyboard.addListener( keyboardHideEvent, () => {
-			if ( previousKeyboardVisible !== false ) {
+			if ( isCurrent && previousKeyboardVisible !== false ) {
 				setKeyboardVisible( false );
 			}
 		} );
 		return () => {
+			isCurrent = false;
 			showListener.remove();
 			hideListener.remove();
```

Comment:
The removal of the listeners appears to not cancel execution of any currently registered callback. There were times where `setKeyboardVisible` as called on an unmounted component, resulting in React warnings. The addition of `isCurrent` prevents state updates to the component if it is no longer current.